rdd/api-versioning.md

```diff
@@ -0,0 +1,66 @@
+
+[Using the Accept Header to version your API](http://labs.qandidate.com/blog/2014/10/16/using-the-accept-header-to-version-your-api/) expands a bit more on this particular approach, and explains:
+> ```
+> GET /jobs HTTP/1.1
+> Host: api.example.com
+> Accept: application/vnd.example.api+json;version=2
+> ```
+> 
+> The vnd. part is dictated by [rfc4288-3.2](https://tools.ietf.org/html/rfc4288#section-3.2), and is used to tell standard media types apart from custom ones. A vnd.* media type can be submitted to IANA to be registered as official media type. In theory, you could just use application/json here and add the version parameter, but as [the json standard](https://tools.ietf.org/html/rfc4627#section-6) doesn't allow any parameters, it wouldn't be correct to use that.
+
+In [API Versioning Has No “Right Way”](https://blog.apisyouwonthate.com/api-versioning-has-no-right-way-f3c75457c0b7) 
+Phil Sturgeon goes over some of the most popular choices and then proposes _API Evolution_:
+
+> API Evolution is the concept of never breaking your contracts until you absolutely absolutely have to, then when you do you manage that change with sensible warnings to clients. It is not about making arbitrary changes and breaking stuff.
+>
+> [...] when backwards compatible issues absolutely could not at all be avoided, we took advantage of the fact that a business name for a concept had changed, and took the chance to make our API match the business name.
+
+This approach is very limited though, and may not be applicable in many situations.
+
+## Well Known Cases
+
+- AWS uses dates for versions, places them in the URL, reserves the right to change the default version freely
+- [GitHub](https://developer.github.com/v3/media/#request-specific-version) uses `beta` and `v1` through `v4` for versions, places them in the `Accept` header and reserves the right to change the default version freely  
+- [Facebook's marketing API](https://developers.facebook.com/docs/marketing-api/versions/) uses semver-like tags such as `v2.2` and places them in the URL
+
+## Team's Experience
+
+Everyone in the team has experience or knows a bit about the version-in-url approach.
+
+## Alternatives
+
+In the future we could do more research and put more thought on this topic. Some approaches to keep in mind:
+
+- Supporting an _endpoint feature_ system, rather than global or endpoint versioning.
+- For authenticated requests, storing the version of the API to use in the database for each user, allowing them to change it from an administration panel. 
+
+## Conclusion
+
+Following GitHub's approach, placing the version in the `Accept`header is the best option at the moment. It suits our current needs, requires no changes to the URLs, is already well documented and widely used.
+
+We will tag the current version as `v1`, and it will be the default initially and for a period of time. Once `v2` has been out for a while, the default will be switched to it, and so forth.
```

Comment:
Current version is actually referred to as 0.1 from the beta period. Maybe we add the `claimId` in GET /works and release that as v1?

If no version is specified in the `Accept` header we default to 0.1.

Reply:
Good catch. Fixed now.